binfmt/libelf/libelf_coredump.c

```diff
@@ -327,51 +524,58 @@ static void elf_emit_program_header(FAR struct elf_dumpinfo_s *cinfo,
 
 int elf_coredump(FAR struct elf_dumpinfo_s *cinfo)
 {
-  int segs = 0;
-  int i;
+  int memsegs = 0;
+  int stksegs;
 
-  /* Check the memory region */
+  if (cinfo->pid != INVALID_PROCESS_ID)
```

Comment:
let's hold critical section here since this function may be called from coredump application now.

Reply:
Done